src/session/session_api.c

```diff
@@ -295,9 +295,23 @@ __wt_session_close_internal(WT_SESSION_IMPL *session)
 {
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
+#ifdef HAVE_CALL_LOG
+    bool internal_session;
+#endif
 
     conn = S2C(session);
 
+#ifdef HAVE_CALL_LOG
+    internal_session = F_ISSET(session, WT_SESSION_INTERNAL);
+    if (!internal_session)
+        /*
+         * The call log entry for the session_close API is generated by two functions. The reason is
+         * that the first function (called below) requires session's variables and the second
+         * function (called at the end) requires the ret value.
+         */
+        WT_TRET(__wt_call_log_close_session(session));
```

Comment:
I think only the session address is what's required for the call log entry here. If we want to combine the `__wt_call_log` functions we could save the address in a local variable in pass it in at the end.

Reply:
No we cannot pass it at the end, as we need `conn` from `session`. However, at the end the `conn` reference from `session` is removed. In other words, `session->conn = NULL`.

Reply:
Fair enough, that makes sense.